src/quarto/ai/basic.go

```diff
@@ -62,14 +68,20 @@ func GetNonWinningPiecesListFromState(currentState state.State) []int {
 			}
 		}
 	}
-	var piecesListNonWinning = underscore.Select(piecesListInitial, func(ni int, _ int) bool {
-		var indexWinningPiece = underscore.FindIndex(piecesListWinning, func(nw int, _ int) bool {
-			return ni == nw
-		})
-		return indexWinningPiece < 0
+	return grid.GetListPieceAMinusListPieceB(piecesListInitial, piecesListWinning)
+}
+
+func GetLoosingBoxList(currentState state.State) []grid.Point {
+	boxList := grid.GetEmptyBoxes(currentState.Grid)
+	loosingBoxList := underscore.Select(boxList, func(ni grid.Point, _ int) bool {
+		testState := state.CopyState(currentState)
+		testState.Grid[ni.Y][ni.X] = testState.Piece
+		testState.Piece = 0
+		nonWinningPieces := GetNonWinningPiecesListFromState(testState)
+		return len(nonWinningPieces) == 0
 	})
-	if piecesListNonWinning == nil {
-		return []int{}
+	if loosingBoxList == nil {
+		return []grid.Point{}
 	}
-	return piecesListNonWinning.([]int)
-}
+	return loosingBoxList.([]grid.Point)
+}
```

Comment:
You should have a rule in vscode to add automatically an empty line at the end of each file.